src/ErrorHandler.php

```diff
@@ -426,14 +463,17 @@ private function handleException(\Throwable $exception): void
 
         $previousExceptionHandlerException = $exception;
 
-        // Unset the previous exception handler to prevent infinite loop in case
-        // we need to handle an exception thrown from it
+        // Sets the previous handler so that the exception is printed on the
+        // screen the next time this method runs. This also breaks the infinite
+        // loop that would happen if the previous handler thrown a new exception
         $previousExceptionHandler = $this->previousExceptionHandler;
-        $this->previousExceptionHandler = null;
+        $this->previousExceptionHandler = [$this, 'printException'];
```

Comment:
This doesn't seem transparent to me.

Reply:
This is going to do the same thing that the native handler would do which is printing the exception. We have to do this because in case if there is a `$previousExceptionHandler` set at the first run we call it, so nothing gets printed. At the second run instead we do want to print the error. Also, this is needed to break the infinite loop in case `$previousExceptionHandler` throws an exception

Reply:
I agree that breaking the loop is needed, but "stealing" the job from the native handler seems asking for trouble to me. We need to give the flow back to whatever is happening, not mimic it.

Reply:
It's a lot more complicated than what it seems, because there are a lot of different cases that should be handled and there doesn't seems to be a solution that works for all of them without some sort of hack/workaround. If we don't print the exception ourselves then there may be cases where it doesn't get printed at all. Given that this code is again based on the SF error handler, although with some changes, and that I'm pretty sure that the people that worked on it discovered the same things, I firmly believe that we either choose to take some compromises or we don't fix the fact that an exception thrown inside the next handler after our has been called is not catched

Reply:
I agree. I tend to think that we can leave behind the case of "another handler messed up" if it forces us to implement dirty hacks. It wouldn't be our fault, and we can't cover 100% of the cases.